feature-libs/product-configurator/rulebased/components/attribute/types/radio-button/configurator-attribute-radio-button.component.ts

```diff
@@ -1,147 +1,41 @@
-  extends ConfiguratorAttributeBaseComponent
+  extends ConfiguratorAttributeSingleSelectionBaseComponent
   implements OnInit {
   attributeRadioButtonForm = new FormControl('');
-  loading$ = new BehaviorSubject<boolean>(false);
 
-  @Input() attribute: Configurator.Attribute;
-  @Input() ownerKey: string;
-
-  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+  ngOnInit(): void {
+    this.attributeRadioButtonForm.setValue(this.attribute.selectedSingleValue);
+  }
 
-  // TODO(#11681): make quantityService a required dependency
   /**
-   * default constructor
-   * @param {ConfiguratorAttributeQuantityService} quantityService
+   TODO(issue: #11238): update @deprecated level to the release we are publishing with,
+   It is still 3.1 only because app.module.ts states that we are on 3.1.
+   Finally we must have 3.x, x>=2 here
    */
-  // eslint-disable-next-line @typescript-eslint/unified-signatures
-  constructor(quantityService: ConfiguratorAttributeQuantityService);
-
   /**
-   * @deprecated since 3.3
+   * @deprecated since 3.1
+   * User better onSelect('')
    */
-  constructor();
-
-  constructor(
-    protected quantityService?: ConfiguratorAttributeQuantityService
-  ) {
-    super();
-  }
-
-  ngOnInit(): void {
-    this.attributeRadioButtonForm.setValue(this.attribute.selectedSingleValue);
-  }
-
-  get withQuantity() {
-    return (
-      this.quantityService?.withQuantity(
-        this.attribute.dataType,
-        this.attribute.uiType
-      ) ?? false
-    );
-  }
-
-  get disableQuantityActions() {
-    return (
-      this.quantityService?.disableQuantityActions(
-        this.attribute.selectedSingleValue
-      ) ?? true
-    );
-  }
-
-  onSelect(value: string): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: value,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
   onDeselect(): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        selectedSingleValue: '',
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onHandleQuantity(quantity): void {
-    this.loading$.next(true);
-
-    const event: ConfigFormUpdateEvent = {
-      changedAttribute: {
-        ...this.attribute,
-        quantity,
-      },
-      ownerKey: this.ownerKey,
-      updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
-    };
-
-    this.selectionChange.emit(event);
-  }
-
-  onChangeQuantity(eventObject: any): void {
-    if (!eventObject) {
-      this.onDeselect();
-    } else {
-      this.onHandleQuantity(eventObject);
-    }
+    this.onSelect('');
```

Comment:
Can we add some flag in `onSelect` to indicate it is deselect or not (by default it is not deselect)? Then in `onSelect` function, we check this flag to set value. It will be more readable than this 'this.onSelect('');`, right?

Reply:
I don't think that it will make `this.onSelect('');` more readable if I add a flag to the method that signals what action takes place.